iaso/api/mobile/entity.py

```diff
@@ -117,20 +100,29 @@ class MobileEntityViewSet(ModelViewSet):
     remove_results_key_if_paginated = True
     filter_backends = [filters.OrderingFilter, DjangoFilterBackend, DeletionFilterBackend]
     pagination_class = LargeResultsSetPagination
+    permission_classes = [permissions.IsAuthenticated, HasPermission("menupermissions.iaso_entities")]
     lookup_field = "uuid"
 
     def get_serializer_class(self):
         return MobileEntitySerializer
 
     def get_queryset(self):
-        profile = self.request.user.iaso_profile
+        user = self.request.user
+        app_id = self.request.query_params.get("app_id")
+
+        if not app_id:
+            raise ParseError("app_id is required")
+
+        queryset = Entity.objects.filter_for_user_and_app_id(user, app_id)
 
-        # This function alter the queryset by adding non_deleted_instances
-        queryset = filter_queryset_for_mobile_entity(Entity.objects.filter(account=profile.account), self.request)
+        filter_for_mobile_entity(queryset, self.request)
 
         # we give all entities having an instance linked to the one of the org units allowed for the current user
-        orgunits = OrgUnit.objects.hierarchy(profile.org_units.all())
-        if orgunits:
-            queryset = queryset.filter(instances__org_unit__in=orgunits)
+        if queryset and user and user.is_authenticated:
+            orgunits = OrgUnit.objects.hierarchy(user.iaso_profile.org_units.all())
+            if orgunits and len(orgunits) > 0:
+                queryset = queryset.filter(instances__org_unit__in=orgunits)
+            else:
+                queryset = queryset.none()
```

Comment:
I am just wondering if the condition should not be:
```suggestion
            if orgunits and len(orgunits) == 0:
                return queryset.none()
            
            if orgunits:
                queryset = queryset.filter(instances__org_unit__in=orgunits)
```
Because I believe that when a user is not assigned to any orgunits, then they have access to everything.

Reply:
Je dois faire un test pour voir ce qu'il retourne quand on assigne rien au user